lib/ansible/modules/packaging/os/redhat_repository.py

```diff
@@ -0,0 +1,275 @@
+#!/usr/bin/python
+
+# Giovanni Sciortino (giovannibattistasciortino@gmail.com)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+
+DOCUMENTATION = '''
+---
+module: redhat_repository
+short_description: Manage RHSM repositories using the C(subscription-manager) command
+description:
+    - Manage(List/Enable/Disable) RHSM repositories to the Red Hat Subscription Management entitlement platform using the C(subscription-manager) command
+version_added: "2.4"
+author: "Giovanni Sciortino (giovannibattistasciortino@gmail.com)"
+notes:
+    - In order to manage rhsm repositories the system must be already registered to rhsm manually or using the ansible module redhat_subscription.
+requirements:
+    - subscription-manager
+options:
```

Comment:
I think it would be much more Ansible-like if there was instead a `name` option that could take multiple names and then a `state` option that could be set to `list`, `enabled`, or `disabled`. That would also eliminate the need for `disable_first` because you would instead run this module twice to get the desired effect. When set to `list`, name could be a not required parameter.